From 192c1f4821c4f8bb2229d0f2983b7ef674e3860e Mon Sep 17 00:00:00 2001
From: Colin McCabe <cmccabe@apache.org>
Date: Fri, 25 Jul 2014 22:13:03 +0000
Subject: [PATCH 260/791] HDFS-6755. There is an unnecessary sleep in the code
 path where DFSOutputStream#close gives up its
 attempt to contact the namenode (mitdesai21 via
 cmccabe)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-2@1613523 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 70a179d29ff495445e06ef7f1098faedbd5a1d78)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt
---
 .../org/apache/hadoop/hdfs/DFSOutputStream.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
index fda4954..b27c24e 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
@@ -2142,12 +2142,12 @@ private void completeFile(ExtendedBlock last) throws IOException {
             throw new IOException(msg);
         }
         try {
-          Thread.sleep(localTimeout);
           if (retries == 0) {
             throw new IOException("Unable to close file because the last block"
                 + " does not have enough number of replicas.");
           }
           retries--;
+          Thread.sleep(localTimeout);
           localTimeout *= 2;
           if (Time.now() - localstart > 5000) {
             DFSClient.LOG.info("Could not complete " + src + " retrying...");
-- 
1.7.9.5

